### ISSUE TITLE
[terra-framework] enhance Keyboard Focus styles to pass '2.4.11 Focus Visible'

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-framework#1262]
# Feature Request

## Description

WCAG 2.1 (Level AA) introduced criterion [2.4.11: Focus Visible (Enhanced)](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced) which raises the WCAG 2 (Level A) [2.4.7: Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) criterion standard of when a user interface element displays a visible keyboard focus that it's color contrast ratio be at least 3:1 or thickness of at least 2 CSS pixels.

All interactive terra elements have a visible keyboard focus style present, but not tested to see if they meet automated AXE testing for the 2.4.7 /  2.4.11 criterion. All components in `tera-core` need to be audited and provide updated styling as needed, as well as adding any appropriate additional tests for confirmation of pass/fail.

\
.

## Solution

Audit, Add testing, and UX to provide appropriate changes to all components in terra-framework

Components affected:
- [ ] _TODO: list out each component needing update here..._

## Related Issues

- [terra-toolkit #443](https://github.com/cerner/terra-toolkit/issues/443)
- [terra-core #3174](https://github.com/cerner/terra-core/issues/3174)
- terra-clinical _add issue number_
- terra-application _add issue number_

## Reference Context / Screenshots
_see [terra-core #3174](https://github.com/cerner/terra-core/issues/3174) for reference screenshot_

**Comment by neilpfeiffer** [comment]
Keeping blocked on needing UX + A11Y input. Backlogging as future enhancement.
